test/test.js

```diff
@@ -5,16 +5,53 @@ import { test } from '../src/index.js';
 describe('test', function () {
   it('should detect as path template', function () {
     assert.isTrue(test('/{path}'));
+    // trailing slash is allowed
+    assert.isTrue(test('/{path}/'));
     assert.isTrue(test('/pets/{petId}'));
     assert.isTrue(test('/{entity}/me'));
     assert.isTrue(test('/books/{id}'));
     assert.isTrue(test('/a{test}'));
+    assert.isTrue(test('/{test}a'));
+    // parentheses are allowed
+    assert.isTrue(test('/range({x},{y})'));
+    // repeated parameter names are allowed
+    assert.isTrue(test('/range({x},{y})/secondRange({x},{y})'));
     assert.isTrue(test('/{entity}/{another-entity}/me'));
+    // NOTE: the following scenario should probably be considered invalid, query parameters are not allowed in path templates
     assert.isTrue(test('/pets?offset=0&limit=10'));
+    // -->
     assert.isTrue(test('/'));
+    // special characters in literal are allowed
+    assert.isTrue(test('/-/'));
+    assert.isTrue(test('/~/'));
+    // NOTE: the following scenario should probably be considered invalid, fragments are not allowed in path templates
+    assert.isTrue(test('/#baz'));
+    // NOTE: the following scenarios should probably be considered invalid, should we allow RFC 6570 operators in parameter names?
+    assert.isTrue(test('/{+baz}'));
+    assert.isTrue(test('/{;baz}'));
+    assert.isTrue(test('/{&baz}'));
+    assert.isTrue(test('/{.baz}'));
+    assert.isTrue(test('/{count*}'));
+    assert.isTrue(test('/{y,x}'), '/{y,x}');
+    // NOTE: the following scenarios should probably be considered invalid due to the use of special characters
+    assert.isTrue(test('/{foo:baz}'));
+    assert.isTrue(test('/{=baz}'));
+    assert.isTrue(test('/{$baz}'));
+    assert.isTrue(test('/{~baz}'));
+    // -->
+    assert.isTrue(test('/%20'));
+    assert.isTrue(test('/functions/t_Dist_2T'));
+    assert.isTrue(test('/users/$count'));
+    assert.isTrue(test('/users/delta()'));
+    assert.isTrue(test('/directoryObjects/microsoft.graph.user'));
+    assert.isTrue(test('/applications(appId=\'{appId}\')'));
+    assert.isTrue(test('/communications/onlineMeetings/getAllRecordings(meetingOrganizerUserId=\'@meetingOrganizerUserId\')'));
+    // special characters in parameter names are allowed
+    assert.isTrue(test('/users/{user-id}'));
```

Comment:
we need to add a test case for `[` since this needs to be encoded according to RFC 3986

Reply:
Sure, let's do it.